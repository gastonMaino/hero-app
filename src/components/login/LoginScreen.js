import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({history}) => {

    const {dispatch} = useContext(AuthContext);

    const handleLogin = () =>{
        // history.push('/');

        const lastpath = localStorage.getItem('lastpath') || '/';

        dispatch({
            type: types.login,
            payload: {name:'Gaston'}
        })
        history.replace(lastpath);
    }

    return (
        <main className='main-login'>
            <article className='container'>
                <div className='container-title-login'>
                    <h1 className='title-login'>Login Screen</h1>
                </div>
                <button className='button-primary-login' onClick={handleLogin}>
                    Login
                </button>
            </article>
        </main>
    )
}
