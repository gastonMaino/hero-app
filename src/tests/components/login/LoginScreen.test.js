import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext"
import { LoginScreen } from "../../../components/login/LoginScreen"

describe('tests in <LoginScreen />', () => {
    const historyMock = {
        replace: jest.fn(),
        listen: jest.fn(),
        createHref: jest.fn()
    }    

    const contexValue = {
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={contexValue}>
            <LoginScreen history={historyMock} />
        </AuthContext.Provider>
    )

    test('should match to snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should call dispatch and history replace', () => {
        wrapper.find('button').prop('onClick')()

        expect(historyMock.replace).toHaveBeenCalledWith('/');
        expect(contexValue.dispatch).toHaveBeenCalled();
    })

    test('should call history.replace(lastpath) ', () => {

        localStorage.setItem('lastpath', '/dc');

        wrapper.find('button').prop('onClick')();

        expect(historyMock.replace).toHaveBeenCalledWith('/dc')
    })
    
    
    
    
})

