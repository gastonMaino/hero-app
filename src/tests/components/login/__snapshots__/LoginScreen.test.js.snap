// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <LoginScreen /> should match to snapshot 1`] = `
<main
  className="main-login"
>
  <article
    className="container"
  >
    <div
      className="container-title-login"
    >
      <h1
        className="title-login"
      >
        Login Screen
      </h1>
    </div>
    <button
      className="button-primary-login"
      onClick={[Function]}
    >
      Login
    </button>
  </article>
</main>
`;
